app/src/main/java/de/xikolo/viewmodels/main/ProfileViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package de.xikolo.viewmodels.main
+
+import androidx.lifecycle.LiveData
+import de.xikolo.models.User
+import de.xikolo.models.dao.UserDao
+import de.xikolo.viewmodels.base.BaseViewModel
+import de.xikolo.viewmodels.shared.EnrollmentDelegate
+import de.xikolo.viewmodels.shared.UserDelegate
+
+class ProfileViewModel : BaseViewModel() {
+
+    private val userDelegate = UserDelegate(realm)
+    private val enrollmentDelegate = EnrollmentDelegate(realm)
+
+    private val userDao = UserDao(realm)
+
+    val user: LiveData<User> by lazy {
+        userDao.getCurrent()
+    }
+
+    val enrollments = enrollmentDelegate.enrollments
+
+    val enrollmentCount
+        get() = enrollmentDelegate.enrollmentCount
+
+    override fun onFirstCreate() {
+        userDelegate.requestUserWithProfile(networkState, false)
+        enrollmentDelegate.requestEnrollmentList(networkState, false)
+    }
+
+    override fun onRefresh() {
+        userDelegate.requestUserWithProfile(networkState, true)
+        enrollmentDelegate.requestEnrollmentList(networkState, true)
```

Comment:
I think the enrollment request network state should not be connected with the UI, it is a secondary resource and requested concurrently with the profile.

Reply:
Solved